Snowflake.Data.Tests/SFConnectionIT.cs

```diff
@@ -132,13 +134,75 @@ public void TestIncorrectUserOrPasswordBasicConnection()
                 {
                     // Expected
                     logger.Debug("Failed opening connection ", e);
-                    Assert.AreEqual("08006", e.SqlState); // Connection failure
+                    Assert.AreEqual(ConnectionFailureErrorCode, e.SqlState);
                 }
 
                 Assert.AreEqual(ConnectionState.Closed, conn.State);
 			}
         }
 
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        public void TestConnectionIsNotMarkedAsOpenWhenWasNotCorrectlyOpenedBefore(bool explicitClose)
+        {
+            SnowflakeDbConnectionPool.SetPooling(true);
+            SnowflakeDbConnection snowflakeConnection = null;
+            for (var i = 0; i < 2; ++i)
+            {
+                logger.Debug($"Running try #{i}");
+                try
+                {
+                    snowflakeConnection = new SnowflakeDbConnection(ConnectionStringWithInvalidUserName);
+                    snowflakeConnection.Open();
+                    Assert.Fail("Connection open should fail");
+                }
+                catch (SnowflakeDbException e)
+                {
+                    Assert.AreEqual(ConnectionFailureErrorCode, e.SqlState);
+                    AssertConnectionIsNotOpen(snowflakeConnection);
+                    if (explicitClose)
+                    {
+                        snowflakeConnection.Close();
+                        AssertConnectionIsNotOpen(snowflakeConnection);
+                    }
+                    snowflakeConnection = null;
+                }
+            }
+        }
+
+        [Test]
+        public void TestConnectionIsNotMarkedAsOpenWhenWasNotCorrectlyOpenedWithUsing()

```

Comment:
what do you think of renaming to ... WithUsingClause or ...WithUsingStatement?

Reply:
Used suffix `WithUsingClause`